app/src/main/kotlin/com/simplemobiletools/applauncher/activities/MainActivity.kt

```diff
@@ -126,13 +126,20 @@ class MainActivity : SimpleActivity(), RefreshRecyclerViewListener {
         checkInvalidApps()
         initZoomListener()
         setupAdapter(displayedLaunchers)
+        maybeShowEmptyView(displayedLaunchers)
     }
 
     private fun setupAdapter(launchers: ArrayList<AppLauncher>) {
         AppLauncher.sorting = config.sorting
         launchers.sort()
 
-        LaunchersAdapter(this, launchers, this, launchers_grid) {
+        LaunchersAdapter(
+            activity = this,
+            launchers = launchers,
+            listener = this,
+            recyclerView = launchers_grid,
+            onItemsRemoved = ::onItemsRemoved,
```

Comment:
whats ::?

Reply:
`::` provides a reference to a function like one can get reference to a class like`MyClass::class`

other options:

-  simply pass a lambda like `onItemsRemoved = { // do something here },`
-  or setup a listener class

Reply:
change this please, this is unreadable